files/acl/restrict-unpublished-files.php

```diff
@@ -105,3 +104,43 @@ function get_attachment_id_from_file_path( $path ) {
 
 	return $attachment_id;
 }
+
+/**
+ * PURGE all attachments for a given post.
+ *
+ * This is to allow us to propagate visibility changes ASAP to the edges
+ * and also to allow us to cache public files for much longer (which comes
+ * some performance benefits).
+ *
+ * @param array urls Array of URLs to be purged
+ * @param int post_id The ID of the post for which we're purging URLs
+ *
+ * @return array An array of URLs to be purged
+ */
+function purge_attachments_for_post( $urls, $post_id ) {
+	$post = get_post( $post_id );
+
+	if ( empty( $post ) ) {
+		return $urls;
+	}
+
+	$attachment_ids = get_posts( [
+		'post_parent'    => $post->ID,
+		'post_type'      => 'attachment',
+		'posts_per_page' => 100, // Set a reasonably high limit (instead of -1 as default)
+		'orderby' => 'ID', // For performance (instead of `date` as default)
+		'order' => 'ASC',
+		'fields' => 'ids',
+		'suppress_filters' => false,
+	] );
```

Comment:
This might be a silly comment, what happens if a post has more than 100 attachments, should we make sure to get all of them, possibly in batches?

Reply:
Yes, we can do that. It's a not very likely scenario but I was just being a lazy engineer :)

Reply:
Actually, since we're just fetching IDs, we can probably bump the number much higher. I would say 250 is probably the max. If you have more than that, you have bigger problems :)

Reply:
Thanks for the reply :)

I know it's very unlikely, but since this will leave files marked as public for very long when a post is updated if there are more than 100 I felt compelled to reply ;)

Reply:
I wouldn't say it's "very unlikely", a large gallery can easily be over 100 images.

Reply:
Updated to 250.